### ISSUE TITLE
Generation failing with infinite time for imbalanced datasets

### ISSUE THREAD

**Issue by omaralvarez** [Team-TUD/CTAB-GAN-Plus#22]
First of all, thanks for your contribution. I am using your model for dataset rebalancing, and with datasets with low and imbalanced samples I am facing a problem. Generation fails due to reaching the maximum amount of tries, I have tried several approaches like increasing the epochs or trying to use #7, to no avail, generation never completes. One of the datasets in which is happening is:

https://imbalanced-learn.org/stable/datasets/index.html

``` python
from imblearn.datasets import fetch_datasets

ecoli = fetch_datasets()['ecoli']
ecoli.data.shape
```

**Comment by omaralvarez** [comment]
I think I found the problem, it has to do with negative numeric target labels. Using strings it works. The same thing is happening in your other model CTABGAN.

**Comment by zhao-zilong** [comment]
Hi @omaralvarez thanks for your comment. What do you mean by negative numeric target labels? like "-1" and "-5.78" in the target label column?

**Comment by omaralvarez** [comment]
I finally found the issue, sometimes the original pandas datatypes are not given back when the model generates samples. It returns objects (I think strings). So that was causing a bug in my code. A simple:

``` python
        sample = self.synthesizer.sample(self.batch_size)

        return self.data_prep.inverse_prep(sample).astype(
            dtype=self.raw_df.dtypes.to_dict()
        )
```

Fixes the issue.

**Comment by zhao-zilong** [comment]
OK, cool, thanks @omaralvarez  Would you like to create a pull request to improve this part of code?

**Comment by omaralvarez** [comment]
Yep, no problem. Right now I am tight on time, but as soon as I can I will whip out a pull request.
